docs/useAzkaban.rst

```diff
@@ -245,14 +245,23 @@ executing, several options can be set.
 
 .. image:: figures/executeflowconcurrent.png
 
-Flow Parameters
+Runtime Properties
 ~~~~~~~~~~~~~~~
 
-Allows users to override flow parameters. The flow parameters override
-the global properties for a job, but not the properties of the job
-itself.
+Runtime Properties allow users to override properties temporarily.
 
-.. image:: figures/executeflowparameters.png
+Note: This feature was previously known as Flow Parameters.
+
+.. image:: figures/executeflowruntimeproperties.png
+
+The scope of each runtime property is defined by the Node selection. Rows with `ROOT` Node apply to all jobs in the flow. Properties can be scoped per sub-flow as well as individual jobs. Most specific node path takes precedence.
+
+Overrides are applied depending on the Azkaban server configuration property `executor.props.resolve.overrideExisting.enabled`:
+
++ **If overriding of existing properties is not enabled (default):**
+   + Runtime properties override the global properties for a job, but not the properties of the job itself.
++ **If overriding of existing properties is enabled:**
+   + Runtime properties override also existing job props
```

Comment:
I understand the documentation already referred to "global properties for a job" but I'm not sure what that means  because there are properties defined a flow/rootnode level which apply to all jobs in a DAG but there are also "global" properties at Application(Azkaban) level and plugin level that could impact all jobs as well.

Here is an alternative explanation. Does it sound clearer?
`executor.props.resolve.overrideExisting.enabled=false` (default)
Predefined job properties (props in .job, .properties (Flow 1.0)  or .flow (Flow 2.0) files) have higher precedence than runtime properties overrides specified in any parent node.
`executor.props.resolve.overrideExisting.enabled=true`
Runtime properties overrides in parent nodes have higher precedence than predefined job properties.

Reply:
## enabled=false

> Predefined job properties (props in .job, .properties (Flow 1.0) or .flow (Flow 2.0) files) have higher precedence than runtime properties overrides specified in any parent node.

**I'm afraid that's incorrect**. It's only the job level properties that have higher precedence than runtime properties. An example of that here:

https://github.com/azkaban/azkaban/blob/8df583c350f050c097780cdc8be69c973cea861f/azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerPropertyResolutionTest.java#L62-L67

The flow under test is loaded from https://github.com/azkaban/azkaban/tree/master/test/execution-test-data/execpropstest and especially there is this `.properties` file with the following property:

https://github.com/azkaban/azkaban/blob/8df583c350f050c097780cdc8be69c973cea861f/test/execution-test-data/execpropstest/moo.properties#L3

Now, let's jump into that `assertProperties()`.

For example this ROOT level runtime property is set:

https://github.com/azkaban/azkaban/blob/8df583c350f050c097780cdc8be69c973cea861f/azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerPropertyResolutionTest.java#L120

It's asserted that the runtime property was used (instead of using the value defined in `moo.properties`):

https://github.com/azkaban/azkaban/blob/8df583c350f050c097780cdc8be69c973cea861f/azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerPropertyResolutionTest.java#L159

The order of precedence is also written out as a comment in the same test method:
https://github.com/azkaban/azkaban/blob/8df583c350f050c097780cdc8be69c973cea861f/azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerPropertyResolutionTest.java#L152-L153

## enabled=true

> Runtime properties overrides in parent nodes have higher precedence than predefined job properties.

I feel this somehow lacking because it sounds like it only concerns the job properties, although it's also .properties (or any predefined properties) that are overridden in this case.

How about this:

> Runtime properties (also those that are set on a parent node) have higher precedence than predefined properties.

?

Reply:
> Runtime properties (also those that are set on a parent node) have higher precedence than predefined properties.

Sounds good!

Regarding:
> Predefined job properties (props in .job, .properties (Flow 1.0) or .flow (Flow 2.0) files) have higher precedence than runtime properties overrides specified in any parent node.

I think in Flow 2.0 properties defined in _.properties_ files are interpreted as parent node props instead of job specific props (those defined in _.job_ files). That was my understanding as well for Flow 1.0 DAG definitions but I may be wrong. 
Based on that reasoning, `props5=moo5` would not count as a "predefined job property" so setting it to the runtime specified value it's ok and doesn't invalidate the sentence right?

But let's scratch that, what do you think of:
```
Runtime properties don't override predefined properties from a more specific node path. In particular
runtime properties at flow level don't override predefined properties at job level.
```
instead?

Reply:
That's perfect. Using that.